lib/focus/focus_interface.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// A component or directive that can be programmatically focused.
+///
+/// Directive can manage if it means to put focus on root of itself
```

Comment:
"Directive can manage if it means to ..." I'm not quite sure what that means. Maybe just suggest that "focus is put on this component or directive, or meaningful component inside."